clientv3/watch.go

```diff
@@ -367,7 +367,7 @@ func (w *watcher) closeStream(wgs *watchGrpcStream) {
 }
 
 func (w *watchGrpcStream) addSubstream(resp *pb.WatchResponse, ws *watcherStream) {
-	if resp.WatchId == -1 {
+	if resp.Canceled && resp.CancelReason != "" {
```

Comment:
@xiang90 This PR breaks backward compatibility if we want to returning the original ID from watch create request.

<= v3.3: canceled watch request response always `resp.WatchId == -1`
>= v3.4: canceled watch request response may have watch ID.

So, the way to check cancellation changes. We could keep returning -1 as watch ID on cancellation, but clients won't be able to know which one is canceled...